src/Grids/grid_utils.jl

```diff
@@ -1,9 +1,12 @@
 using CUDA
+using Printf
 
 #####
 ##### Convenience functions
 #####
 
+on_architecture(::Nothing, grid) = grid
```

Comment:
Docs spit error: 

```
MethodError: on_architecture(::Nothing, ::RectilinearGrid{Float64, Bounded, Flat, Bounded, Float64, Float64, Float64, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}, CPU}) is ambiguous. Candidates:

on_architecture(::Nothing, grid) in Oceananigans.Grids at /var/lib/buildkite-agent/builds/tartarus-3/clima/oceananigans/src/Grids/grid_utils.jl:8

on_architecture(new_arch, old_grid::RectilinearGrid) in Oceananigans.Grids at /var/lib/buildkite-agent/builds/tartarus-3/clima/oceananigans/src/Grids/rectilinear_grid.jl:416

Possible fix, define

 on_architecture(::Nothing, ::RectilinearGrid)
```

```suggestion
on_architecture(::Nothing, grid::AbstractGrid) = grid
```

?

Reply:
```Julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.6.5 (2021-12-19)
 _/ |\__'_|_|_|\__'_|  |
|__/                   |

julia> f(::Nothing, a) = a
f (generic function with 1 method)

julia> f(x, a::Float64) = a+x
f (generic function with 2 methods)

julia> f(nothing, 2)
2

julia> f(nothing, 2.0)
ERROR: MethodError: f(::Nothing, ::Float64) is ambiguous. Candidates:
  f(::Nothing, a) in Main at REPL[1]:1
  f(x, a::Float64) in Main at REPL[2]:1
Possible fix, define
  f(::Nothing, ::Float64)
Stacktrace:
 [1] top-level scope
   @ REPL[7]:1
```


but

```Julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.6.5 (2021-12-19)
 _/ |\__'_|_|_|\__'_|  |
|__/                   |

julia> f(x, a::Float64) = a+x
f (generic function with 1 method)

julia> f(::Nothing, a::Float64) = a
f (generic function with 2 methods)

julia> f(nothing, 2.0)
2.0
```